independent-projects/tools/common/src/main/java/io/quarkus/devtools/project/buildfile/MavenBuildFile.java

```diff
@@ -0,0 +1,121 @@
+package io.quarkus.devtools.project.buildfile;
+
+import static io.quarkus.devtools.project.extensions.Extensions.toKey;
+
+import io.quarkus.bootstrap.model.AppArtifactCoords;
+import io.quarkus.bootstrap.model.AppArtifactKey;
+import io.quarkus.devtools.project.BuildTool;
+import io.quarkus.maven.utilities.MojoUtils;
+import io.quarkus.platform.descriptor.QuarkusPlatformDescriptor;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+
+public class MavenBuildFile extends BuildFile {
+
+    private AtomicReference<Model> modelRef = new AtomicReference<>();
+
+    public MavenBuildFile(final Path projectFolderPath, final QuarkusPlatformDescriptor platformDescriptor) {
+        super(projectFolderPath, platformDescriptor);
+    }
+
+    @Override
+    public void writeToDisk() throws IOException {
+        if (getModel() == null) {
+            return;
+        }
+        try (ByteArrayOutputStream pomOutputStream = new ByteArrayOutputStream()) {
+            MojoUtils.write(getModel(), pomOutputStream);
+            writeToProjectFile(BuildTool.MAVEN.getDependenciesFile(), pomOutputStream.toByteArray());
+        }
+    }
+
+    @Override
+    protected void addDependencyInBuildFile(AppArtifactCoords coords) throws IOException {
+        if (getModel() != null) {
+            final Dependency d = new Dependency();
+            d.setGroupId(coords.getGroupId());
+            d.setArtifactId(coords.getArtifactId());
+            d.setVersion(coords.getVersion());
+            // When classifier is empty, you get  <classifier></classifier> in the pom.xml
+            if (coords.getClassifier() != null && !coords.getClassifier().isEmpty()) {
```

Comment:
@aloubyansky maybe we should change the AppArtifactKey/Coords instead, having empty string classifier seems not working with the maven api?

Reply:
What do you mean? The only reason we use an empty classifier is because this is how Maven does it.

Reply:
We discussed on zulip and concluded that maven api is behavior differently for resolving artifact and dependencies. Making sure we don't put an empty classifier in the dependency prevent from having `<classifier></classifier>` in the pom.xml.